dissect/target/volume.py

```diff
@@ -53,13 +54,35 @@ def __init__(
         self.serial = serial
         self._volumes_list: list[Volume] = None
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"<{self.__class__.__name__} serial={self.serial}>"
 
+    @classmethod
+    def detect(cls, fh: BinaryIO) -> bool:
+        """Detects whether this ``VolumeSystem`` class can be opened on the given file-like object.
+
+        The position of ``fh`` will be restored before returning.
+
+        Returns:
+            ``True`` or ``False`` if the ``VolumeSystem`` can be opened from this file-like object.
+        """
+        offset = fh.tell()
+        try:
+            return cls._detect(fh)
```

Comment:
Shouldn't it do an `fh.seek(0)` first? Otherwise the docstring of `_detect()` is incorrect.

And I noiticed not all `detect()`'s originally did this, that won't be a problem?

Reply:
Oops, yeah forgot to add it.